nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/controllers/nf-ng-canvas-flow-status-controller.js

```diff
@@ -400,6 +406,554 @@
+                                    version: ruleInfo.bundle.version
+                                })).done(function () {});
+                            });
+                            unbindRuleMenuHandling();
+                        };
+
+                        function unbindRuleMenuHandling() {
+                            $('#rule-menu-more-info').unbind('click', openRuleMoreInfoDialog);
+                            $('#rule-menu-edit-rule').unbind('click', openRuleDetailsDialog);
+                            $(document).unbind('click', closeRuleWindow);
+                        }
+
+                    });
+                },
+
+                /**
+                 * Set event bindings for violation menus
+                 */
+                setViolationMenuHandling: function(response, groupId) {
+                    $('.violation-menu-btn').click(function(event) {
+                        // stop event from immediately bubbling up to document and triggering closeViolationWindow
+                        event.stopPropagation();
+                        var violationInfo = $(this).data('violationInfo');
+                        $('#rule-menu').hide();
+                        $('#violation-menu').show();
+                        $('#violation-menu').position({
+                            my: "left top",
+                            at: "left top",
+                            of: event
+                        });
+
+                        // violation menu bindings
+                        $('#violation-menu-more-info').on( "click", openRuleMoreInfoDialog);
+                        $('#violation-menu-go-to').on('click', goToComponent);
+                        $(document).on('click', closeViolationWindow);
+
+                        function closeViolationWindow(e) {
+                            if ($(e.target).parents("#violation-menu").length === 0) {
+                                $("#violation-menu").hide();
+                                unbindViolationMenuHandling();
+                            }
+                        }
+
+                        function openRuleMoreInfoDialog() {
+                            var rule = response.rules.find(function(rule){ 
+                                return rule.id === violationInfo.ruleId;
+                            })
+                            $('#violation-menu').hide();
+                            $('#violation-type-pill').empty()
+                                                    .removeClass()
+                                                    .addClass(violationInfo.enforcementPolicy.toLowerCase() + ' violation-type-pill')
+                                                    .append(violationInfo.enforcementPolicy);
+                            $('#violation-display-name').empty().append(violationInfo.violationMessage);
+                            $('#violation-description').empty().append(rule.descriptors['component-type'].description);
```

Comment:
The same issue here as above:
This refers to the description of the PropertyDescriptor named 'component-type'.
That is very specific to the DisallowComponentType rule.
If a rule doesn't have that descriptor this fails with an error.

But in general I don't think adding the Rule's PropertyDescriptors'  info to the rule violation info dialog page is desirable.